File: crates/auto-lang/test/a2c/014_hetero_enum/hetero_enum.expected.c
#include "hetero_enum.h"


int main(void) {
    struct Atom atom = {.tag = ATOM_INT, .as.Int = 11};

    switch (atom.tag) {
    case ATOM_INT:
        {
            int i = atom.as.Int;
            {
                printf("%s %d\n", "Got Int:", atom.as.Int);
            }
            break;
        }
    case ATOM_CHAR:
        {
            char c = atom.as.Char;
            {
                printf("%s %c\n", "Got Char:", atom.as.Char);
            }
            break;
        }
    case ATOM_FLOAT:
        {
            float f = atom.as.Float;
            {
                printf("%s %f\n", "Got Float:", atom.as.Float);
            }
            break;
        }
    }
    return 0;
}

File: crates/auto-lang/test/a2ts/06_pattern_matching/001_hetero_enum/hetero_enum.expected.ts
export type Atom =
    { _tag: "Int", value: number }
    | { _tag: "Char", value: number }
    | { _tag: "Float", value: number };

export const Atom = {
    Int: (value: number) => ({ _tag: "Int" as const, value }),
    Char: (value: number) => ({ _tag: "Char" as const, value }),
    Float: (value: number) => ({ _tag: "Float" as const, value })
};


function main(): void {
    const atom = Atom.Int(11);
    

        const __auto_is_0 = atom;
    if (__auto_is_0._tag === "Int") {
        const i = __auto_is_0.value;
        console.log("Got Int:", i);
    }
     else if (__auto_is_0._tag === "Char") {
        const c = __auto_is_0.value;
        console.log("Got Char:", c);
    }
     else if (__auto_is_0._tag === "Float") {
        const f = __auto_is_0.value;
        console.log("Got Float:", f);
    }
}

main();
